src/main/java/macrobase/analysis/pipeline/BasicBatchedPipeline.java

```diff
@@ -57,15 +59,22 @@ public AnalysisResult contextualAnalyze(DataIngester ingester) throws SQLExcepti
     AnalysisResult run() throws SQLException, IOException, ConfigurationException {
         long startMs = System.currentTimeMillis();
         DataIngester ingester = conf.constructIngester();
+        // TODO: this should be a new pipeline
+        // Needs to happen early before ingester is possibly consumed. Downstream stages are allowed to drain
+        // upstream stages in construction.
+        if(contextualEnabled){
+            return contextualAnalyze(ingester);
+        }
+
         TimedBatchIngest batchIngest = new TimedBatchIngest(ingester);
         FeatureTransform featureTransform = new BatchScoreFeatureTransform(conf, batchIngest, conf.getTransformType());
         OutlierClassifier outlierClassifier = new BatchingPercentileClassifier(conf, featureTransform);
+        if (conf.getBoolean(MacroBaseConf.CLASSIFIER_DUMP, MacroBaseDefaults.CLASSIFIER_DUMP)) {
```

Comment:
Ha, not my favorite hack, but a reasonable compromise. @mamikonyana I think this addresses your concern too, right?

Any chance we can do this in the `MacroBaseConf` object instead?